src/main/java/com/marklogic/kafka/connect/source/MarkLogicConstraintValueStore.java

```diff
@@ -0,0 +1,82 @@
+package com.marklogic.kafka.connect.source;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.document.DocumentDescriptor;
+import com.marklogic.client.document.JSONDocumentManager;
+import com.marklogic.client.ext.util.DefaultDocumentPermissionsParser;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import org.springframework.util.StringUtils;
+
+import java.util.Map;
+
+public class MarkLogicConstraintValueStore extends ConstraintValueStore {
+    private final DefaultDocumentPermissionsParser permissionsParser = new DefaultDocumentPermissionsParser();
+    private final DatabaseClient databaseClient;
+    private final String constraintStorageUri;
+    private String constraintStoragePermissions;
+    private String constraintStorageCollections;
+
+    MarkLogicConstraintValueStore(DatabaseClient databaseClient, String constraintStorageUri, String constraintColumn,
+                                  Map<String, Object> parsedConfig) {
+        super(constraintColumn);
+        this.databaseClient = databaseClient;
+        this.constraintStorageUri = constraintStorageUri;
+        constraintStoragePermissions = (String) parsedConfig.get(MarkLogicSourceConfig.CONSTRAINT_STORAGE_PERMISSIONS);
+        if (!StringUtils.hasText(constraintStoragePermissions)) {
+            constraintStoragePermissions = null;
+        }
+        constraintStorageCollections = (String) parsedConfig.get(MarkLogicSourceConfig.CONSTRAINT_STORAGE_COLLECTIONS);
+        if (!StringUtils.hasText(constraintStorageCollections)) {
+            constraintStorageCollections = null;
+        }
+    }
+
+    public void storeConstraintState(String previousMaxConstraintColumnValue, int lastRowCount) {
+        String constraintStateJson = "";
+        try {
+            constraintStateJson = buildConstraintState(previousMaxConstraintColumnValue, lastRowCount);
+            JSONDocumentManager JSONDocMgr = databaseClient.newJSONDocumentManager();
+            StringHandle handle = new StringHandle(constraintStateJson);
```

Comment:
This should use `.withFormat(Format.JSON)` so that ML knows to create a JSON document as opposed to relying on the user-defined URI.